test/TestApps/ToastNotificationsDemoApp/main.cpp

```diff
@@ -0,0 +1,202 @@
+            {
+                LOG_HR_MSG(
+                    args.extendedError,
+                    "The channel request is in back-off retry mode because of a retryable error! Expect delays in acquiring it. RetryCount = %d",
+                    args.retryCount);
+            }
+        });
+
+    auto result = co_await channelOperation;
+
+    if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+    {
+        auto channelUri = result.Channel().Uri();
+
+        std::cout << "channelUri: " << winrt::to_string(channelUri.ToString()) << std::endl << std::endl;
+
+        auto channelExpiry = result.Channel().ExpirationTime();
+
+        // Register Push Event for Foreground
+        winrt::event_token token = result.Channel().PushReceived([](const auto&, PushNotificationReceivedEventArgs const& args)
+            {
+                auto payload = args.Payload();
+
+                // Do stuff to process the raw payload
+                std::string payloadString(payload.begin(), payload.end());
+                std::cout << "Push notification content received from FOREGROUND: " << payloadString << std::endl << std::endl;
+                args.Handled(true);
+            });
+        // Caller's responsibility to keep the channel alive
+        co_return result.Channel();
+    }
+    else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+    {
+        LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with channel request!");
+        co_return nullptr;
+    }
+    else
+    {
+        LOG_HR_MSG(result.ExtendedError(), "Some other failure occurred.");
+        co_return nullptr;
+    }
+
+};
+
+winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel RequestChannel()
+{
+    auto task = RequestChannelAsync();
+    if (task.wait_for(std::chrono::seconds(300)) != AsyncStatus::Completed)
+    {
+        task.Cancel();
+        return nullptr;
+    }
+
+    auto result = task.GetResults();
+    return result;
+}
+
+std::wstring GetEnumString(ToastNotificationSetting const& setting)
+{
+    static std::map<ToastNotificationSetting, std::wstring> enumMapping = {
+        { ToastNotificationSetting::Enabled,  L"Enabled"},
+        { ToastNotificationSetting::DisabledForApplication, L"DisabledForApplication" },
+        { ToastNotificationSetting::DisabledForUser, L"DisabledForUser"},
+        { ToastNotificationSetting::DisabledByGroupPolicy, L"DisabledByGroupPolicy"},
+        { ToastNotificationSetting::DisabledByManifest, L"DisabledByManifest"}
+    };
+    return enumMapping[setting];
+}
+
+int main()
+{
+    // Retrieve the app scenario.
+    bool isPackaged{ Test::AppModel::IsPackagedProcess() };
+    if (!isPackaged)
+    {
+        // Major.Minor version, MinVersion=0 to find any framework package for this major.minor version
+        const UINT32 c_Version_MajorMinor{ 0x00040001 };
```

Comment:
Is this still valid? Should we need to update this in a frequent basis?

Reply:
Because this is a test package it will always be the same major/minor version.